src/authentication/AuthenticatedRoute.tsx

```diff
@@ -1,23 +1,48 @@
-import React from 'react';
+import React, { FC } from 'react';
 import { RouteProps, Redirect, Route } from 'react-router-dom';
+
 import { useAuthentication } from './context';
+import { NotFoundPage } from '../staticPages';
+
+interface AuthenticatedRouteRedirecterProps {
+  requiredPermission?: string;
+}
 
-const AuthenticatedRouteRedirecter = ({ children }: { children: React.ReactNode }) => {
-  const { token } = useAuthentication();
+const AuthenticatedRouteRedirecter: FC<AuthenticatedRouteRedirecterProps> = ({
+  requiredPermission,
+  children,
+}) => {
+  const { token, hasPermission } = useAuthentication();
   const authenticated = !!token;
 
   if (!authenticated) {
     return <Redirect to="/login" />;
   }
 
+  if (requiredPermission && !hasPermission(requiredPermission)) {
+    return <NotFoundPage />;
+  }
+
   return <>{children}</>;
 };
 
-export const AuthenticatedRoute = ({ children, ...rest }: RouteProps) => {
+interface AuthenticatedRouteProps extends RouteProps {
+  requiredPermission?: string;
```

Comment:
Left it as single for now, as we don't have an use case for multiple required permissions yet. When we do, it's easy to change.